tests/zfs-tests/cmd/libzfs_input_check/Makefile.am

```diff
@@ -8,4 +8,5 @@ libzfs_input_check_SOURCES = libzfs_input_check.c
 libzfs_input_check_LDADD = \
 	$(top_builddir)/lib/libspl/libspl.la \
 	$(top_builddir)/lib/libnvpair/libnvpair.la \
-	$(top_builddir)/lib/libzfs_core/libzfs_core.la
+	$(top_builddir)/lib/libzfs_core/libzfs_core.la \
+	$(top_builddir)/lib/libzfs/libzfs.la
```

Comment:
I'm curious why this is required (here and in the nvlist_to_lua Makefile)?

Reply:
The problem was that libzfs_core ended up depending on something in libgeom that was only being resolved through libzfs.
I've verified that the following works.
```
index f81fdc172..617b1cf32 100644
--- a/lib/libzfs_core/Makefile.am
+++ b/lib/libzfs_core/Makefile.am
@@ -14,6 +14,7 @@ libzfs_core_la_LIBADD = \
 
 if BUILD_FREEBSD
 libzfs_core_la_LDFLAGS = -version-info 3:0:0
+libzfs_core_la_LIBADD += -lutil -lgeom
 else
 libzfs_core_la_LDFLAGS = -version-info 1:0:0
 endif
diff --git a/tests/zfs-tests/cmd/libzfs_input_check/Makefile.am b/tests/zfs-tests/cmd/libzfs_input_check/Makefile.am
index b1f330293..32cc46561 100644
--- a/tests/zfs-tests/cmd/libzfs_input_check/Makefile.am
+++ b/tests/zfs-tests/cmd/libzfs_input_check/Makefile.am
@@ -8,5 +8,4 @@ libzfs_input_check_SOURCES = libzfs_input_check.c
 libzfs_input_check_LDADD = \
        $(top_builddir)/lib/libspl/libspl.la \
        $(top_builddir)/lib/libnvpair/libnvpair.la \
-       $(top_builddir)/lib/libzfs_core/libzfs_core.la \
-       $(top_builddir)/lib/libzfs/libzfs.la
+       $(top_builddir)/lib/libzfs_core/libzfs_core.la
```

Reply:
I'll revert the changes to the libs from those two tests.